src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BulkScanContainerClientCache.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.common.implementation.Constants;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.blobrouter.clients.bulkscanprocessor.BulkScanProcessorClient;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.TemporalAccessor;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.azure.storage.common.implementation.Constants.UrlConstants.SAS_EXPIRY_TIME;
+import static com.azure.storage.common.implementation.StorageImplUtils.parseQueryString;
+import static java.time.temporal.ChronoField.INSTANT_SECONDS;
+
+@Service
+public class BulkScanContainerClientCache {
+
+    private final BulkScanProcessorClient bulkScanSasTokenClient;
+    private final Long refreshSasBeforeExpiry;
+
+    private static Cache<String, String> tokenCache;
+
+    public BulkScanContainerClientCache(
+        BulkScanProcessorClient bulkScanSasTokenClient,
+        @Value("${bulk-scan.blob-client.refresh-sas-before-expire-in-sec}") Long refreshSasBeforeExpiry) {
```

Comment:
just in seconds, surely `int` is more than enough

Reply:
oh an `)` in new line

Reply:
operations are in all in long